### ISSUE TITLE
How can I reset the password?

### ISSUE THREAD

**Issue by miguelglopes** [gotify/server#371]
**Have you read the documentation?**
- [x] Yes, but it does not include related information regarding my question.
- [ ] Yes, but the steps described in the documentation do not work on my machine.
- [ ] Yes, but I am having difficulty understanding it and wants clarification.

**You are setting up gotify in**
- [x] Docker
- [ ] Linux native platform
- [ ] Windows native platform

**Describe your problem**
I have root access to the machine where gotify is running, but i forgot my user and password. How can I reset them?

**Comment by jmattheis** [comment]
You can connect to your database directly and set a new password hash.

**Comment by suy17** [comment]
I have the same problem. How can i connect to the database and set new password?

**Comment by jmattheis** [comment]
Depends on the database, you would have to edit the `password` column inside the `users` table, the password must be hashed via bcrypt. There should be tutorials online on how to change database values.

**Comment by Node815** [comment]
Resetting my password using the Bcrypt method did not work.  I changed it to the new value in the database and wrote to it and restarted the container.

![image](https://github.com/gotify/server/assets/20736534/539adcae-b74f-4707-97c6-94f25f22565b)

Login still fails.  

Should the Salt Count be a specific value?

**Comment by matezoltanfarkas** [comment]
For anyone else having similar issues: my password manager messed up saving the generated password as well and I was locked out.

I made a manual copy of the `gotify.db` and saved it locally. Then, I created a fresh new instance and looked into the new `gotify.db` file manually.
I then copied from the new one's user's table the contents of the `pass` field into the new one, restarted the session and used the default password to log-in.
